roles/pulp/templates/pulp.container.j2

```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Pulp 
+After=network.target
+
+[Container]
+Image={{ pulp_image }}
+ContainerName={{ pulp_container_name }}
+PublishPort={{ pulp_ports }}
+{% for volume in pulp_volumes %}
+Volume={{ volume }}
+{% endfor %}
+Environment=PULP_CONTENT_ORIGIN=http://{{ ansible_hostname }}:8080
```

Comment:
I think this port should be the same as what's used as the external port in your `pulp_ports` variable.

That's now structured as `8080` (the port on the host), which you also use here. Then the port `80` is the one inside the container and that's fixed (because it's part of the container image itself).

My recommendation would be to change the `pulp_ports` variable to something that's more like `pulp_port=8080`. Then you can use `PublishPort={{ pulp_port }}:80` and here:
```suggestion
Environment=PULP_CONTENT_ORIGIN=http://{{ ansible_hostname }}:{{ pulp_port }}
```